### PR TITLE
Add script for running all testbenches

### PR DESCRIPTION
**Pull request by ziyaointl**
Adds a script for running all provided FPGA testbenches.
Also adds some documentation in the checkpoint4 spec since that's where I suspect the script will be most useful.
Suggestions are welcome!

### CODE REVIEW

File: hardware/run_all_sims.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
+from subprocess import run
```
Comment (vighneshiyer) [inline hardware/run_all_sims.py]: Drop the .py extension from the file and make this a script (add "#! /usr/bin/env python3" as the first line).

File: hardware/run_all_sims.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+from os import remove, makedirs, path
+from shutil import rmtree
+from typing import Tuple
+
+RESULT_DIR = 'test_results'
+
+def run_bench(name, cleanup, success_cond):
```
Comment (vighneshiyer) [inline hardware/run_all_sims.py]: Could you add type annotations for the arguments and return value of this function

File: hardware/run_all_sims.py (hunk 1 above)
Comment (ziyaointl) [reply hardware/run_all_sims.py]: `removesuffix` requires python 3.9, so I used `/home/ff/eecs151/miniconda3/bin/python3` instead.

File: hardware/run_all_sims.py (hunk 2 above)
Comment (ziyaointl) [reply hardware/run_all_sims.py]: Sure! Added :)

**Pull request merged**